### PR TITLE
[5.2] Added SoftDelete Acceptance Test Functionality

### PR DESCRIPTION
**Pull request by iamkirkbater**
I've added a trait to check that something was soft deleted in the database for better acceptance tests.

Example Usage:

``` php
<?php

use Illuminate\Foundation\Testing\SoftDeletes;

class MyControllerTest extends TestCase
{

    use SoftDeletes;

    ....

    public function test_delete_a_user_successfully()
    {
        $user = new User();
        $user->name = "Test User";
        $user->save();
        $this->seeInDatabase("users", ["name" => "Test User"]);

        $response = $this->call('delete', '/users/'.$user->id, []);
        $this->assertEquals(HTTP::OK, $response->status());
        $this->seeInDatabase("users", ["name" => "Test User"]);
        $this->seeIsSoftDeletedInDatabase("users", ["name" => "Test User");
    }

```

This thoroughly tests that the api endpoint is working as intended.

I just wrapped the code in a trait and submitted the pull request.  The actual code was written on SO here: http://stackoverflow.com/questions/33117746/laravel-unit-testing-how-to-seeindatabase-soft-deleted-row.

### CODE REVIEW

File: src/Illuminate/Foundation/Testing/SoftDeletes.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            'Found unexpected records in database table [%s] that matched attributes [%s].', $table, json_encode($data)
+        ));
+
+        return $this;
+    }
+
+    /**
+     * Assert that a given where condition matches a soft deleted record.
+     *
+     * @param string $table
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/SoftDeletes.php]: cs

File: src/Illuminate/Foundation/Testing/SoftDeletes.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     * @param array  $data
+     * @param string $connection
+     *
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/SoftDeletes.php]: cs

File: src/Illuminate/Foundation/Testing/SoftDeletes.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+trait SoftDeletes
+{
+    /**
+     * Assert that a given where condition does not matches a soft deleted record.
+     *
+     * @param string $table
+     * @param array  $data
+     * @param string $connection
+     *
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/SoftDeletes.php]: cs

**Comment by sileence** [comment]
This is good :+1: IMO. I like the idea of having small Traits for specific testing purposes.

**Comment by taylorotwell** [comment]
Going to hold off on this for now.

**Comment by michaeldyrynda** [comment]
@taylorotwell any chance you'd reconsider this? It would be handy to be able to easily test this.

**Comment by taylorotwell** [comment]
Can't you already test it by just running the query and asserting against the count returned or something?

**Comment by michaeldyrynda** [comment]
I wound up testing it using `seeInDatabase('table', ['id' => $id])->notSeeInDatabase('table', ['id' => $id, 'deleted_at' => null])` - I had only wondered if it was something that could be wrapped behind a nicer single method, as I'd have thought it's something tested often enough to warrant it.

**Comment by nateritter** [comment]
@taylorotwell Just curious on this one... You've built in alias API methods ( for `notSeeInDatabase`, for instance) to make things easier. Why not include this one? It's reasonable.

**Pull request closed without merging**